app/controllers/search_goods.js

```diff
@@ -0,0 +1,92 @@
+import Ember from "ember";
+import searchModule from "./search_module";
+import _ from "lodash";
+
+export default searchModule.extend({
+  minSearchTextLength: 2,
+  filteredResults: [],
+  displayResults: false,
+
+  onSearchTextChange: Ember.observer("searchText", function() {
+    if (this.get("searchText").length > this.get("minSearchTextLength")) {
+      this.reloadResults();
+    }
+    this.set("filteredResults", []);
+  }),
+
+  reloadResults() {
+    this.hideResults();
+    Ember.run.debounce(this, this.showResults, 500);
+  },
+
+  hideResults() {
+    Ember.run(() => {
+      this.set("displayResults", false);
+    });
+  },
+
+  showResults() {
+    Ember.run(() => {
+      this.set("displayResults", true);
+    });
+  },
+
+  getFilterQuery() {
```

Comment:
Clean class 👍 
I like the dividing the queries in those functions, even if some are static content and could technically all be grouped, it shows me exactly where to edit them and what they are

Reply:
yeah right